### PR TITLE
java7-maven docker container..

### PR DESCRIPTION
**Pull request by narcoticfresh**
this is a very simple (as it should be) java7-maven docker container for jenkins.
i tested the pubkey-feature - it worked fine and I could log in with my pubkey..

also i was able to clone and build the ml3k jar successfully.. hooray!

@hairmare What I'm unsure about is how jenkins clones the repo(?). I tried to write the passed pubkey to  `~/ssh/id_rsa.pub`, but that didn't work. I cloned ml3k via https interactively.. 
Is more work needed so jenkins can clone the repo inside the container?

### CODE REVIEW

**Comment by hairmare** [comment]
Jenkins has the facilities to inject an ssh-agent into a build context.

File: java7-maven/container_files/start-ssh.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+else
+  grep -q "${SSH_PUB_KEY}" ~/.ssh/authorized_keys || echo "${SSH_PUB_KEY} imported_.key" >> ~/.ssh/authorized_keys
+  grep -q "${SSH_PUB_KEY}" ~/.ssh/id_rsa.pub || echo "${SSH_PUB_KEY} imported_.key" > ~/.ssh/id_rsa.pub
+fi
+
+# run forever..
+for (( ; ; ))
+do
+   echo "Pres CTRL+C to stop..."
+   sleep 1
```
Comment (hairmare) [inline java7-maven/container_files/start-ssh.sh]: Sleep moar...

Could you bump this a bit, it's looks awfully low.

Comment (hairmare) [inline java7-maven/container_files/start-ssh.sh]: Or even better, rewrite it to start a non-daemonized sshd process right here...

**Comment by narcoticfresh** [comment]
@hairmare I really think we should let `/etc/init.d/sshd` do what it does.. don't redo whats there already..

But I added a nice process to hang ;-) that should be fine IMHO..

**Comment by hairmare** [comment]
I disagree strongly on this. We should always try to go the docker way and stick to the one-process-per-container approach.

This leads to simpler deploys and leaner containers with a smaller footprint. It also allows commands like ``docker logs`` to work as intended. Since we should call sshd through ``exec`` we also get the added benefit of useable exitcodes.

Look at, say the [fedora/ssh](https://github.com/fedora-cloud/Fedora-Dockerfiles/tree/master/ssh) image for a good example.

File: java7-maven/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM jamesdbloom/docker-java7-maven
```
Comment (hairmare) [inline java7-maven/Dockerfile]: What was wrong with using [library/maven](https://registry.hub.docker.com/_/maven/)? It's probably much closer to upstream.

**Comment by narcoticfresh** [comment]
@hairmare OK; this container also works to build ml3k.. should be more to your gusto now ;-)

**Comment by hairmare** [comment]
@narcoticfresh I did some cleanup in e98b0d9. We should also talk about not running mvn as root. Would that be possible?

**Comment by narcoticfresh** [comment]
@hairmare Thanks for the fixes.. Lot of stuff I didn't realize.. As for root.. Now, of course maven doesn't need to be run as root at all..

**Pull request merged**